directorywatcher/directorywatcher.go

```diff
@@ -0,0 +1,288 @@
+package directorywatcher
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	"gopkg.in/fsnotify.v1"
+
+	"github.com/reddit/baseplate.go/filewatcher"
+	"github.com/reddit/baseplate.go/internal/limitopen"
+	"github.com/reddit/baseplate.go/log"
+)
+
+// DirectoryWatcher loads and parses data from a file and watches for changes to that
+// file in order to refresh it's stored data.
+type DirectoryWatcher interface {
+	// Get returns the latest, parsed data from the DirectoryWatcher.
+	Get() interface{}
```

Comment:
I strongly prefer APIs structured as

```
Parse(dstPtr interface{}) error
```

so that the caller doesn't have to deal with type assertions